Support ECR registry from URL

peter-svensson opened the pull request.

peter-svensson committed:
use go version 1.16

Updated go.mod and github actions to latest go version

argoyle reviewed (changes requested):
Maybe just remove all the commented tests in gcr_test.go? Especially since they seem to be copies of the ECR-tests. 😄

peter-svensson committed:
make sure that correct registry is used

Extract the actual registry id from the ECR URL and use when creating
non-existing repositories.
